docs/guides/grafana-vmgateway-openid-configuration.md

```diff
@@ -0,0 +1,162 @@
+   <img src="grafana-vmgateway-openid-configuration/create-mapper-2.png" width="800">
+   Configure the mapper as follows
+    - `Mapper Type` as `User Attribute`.
+    - `Name` as `vm_access`.
+    - `Token Claim Name` as `vm_access`.
+    - `User Attribute` as `vm_access`.
+    - `Claim JSON Type` as `JSON`.
+      Enable `Add to ID token` and `Add to access token`.
+      <img src="grafana-vmgateway-openid-configuration/create-mapper-3.png" width="800">
+      Click `Save`.
+7. Go to `Users` -> select user to configure claims -> `Attributes`.
+   Specify `vm_access` as `Key`.
+   Specify `{"tenant_id" : {"account_id": 0, "project_id": 0 }}` as `Value`.
+   <img src="grafana-vmgateway-openid-configuration/user-attributes.png" width="800">
+   Click `Save`.
+
+## Configure grafana
+
+It is required to configure Grafana to use OpenID Connect authentication and to forward the JWT token to vmgateway.
+
+Example configuration for Grafana OpenID Connect authentication:
+```ini
+[auth.generic_oauth]
+enabled = true
+allow_sign_up = true
+team_ids =
+allowed_organizations =
+name = keycloak
+client_id = {CLIENT_ID_FROM_IDENTITY_PROVIDER}
+client_secret = {SECRET_FROM_IDENTITY_PROVIDER}
+scopes = openid profile email
+auth_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/auth
+token_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/token
+api_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/userinfo
+```
+
+Start Grafana. You should be able to log in using your identity provider.
+
+Create a datasource in Grafana. For example, Prometheus datasource with the following URL `http://localhost:8431`.
+Enable `Forward OAuth identity` flag.
+<img src="grafana-vmgateway-openid-configuration/grafana-ds.png" width="800">
+
+## Start vmgateway
+
+Now starting vmgateway to enable authentication is as simple as adding the `-enable.auth=true` flag.
+In order to enable multi-tenant access, you must also specify the `-clusterMode=true` flag.
+
+```console
+./bin/vmgateway -eula \
+    -enable.auth=true \
+    -clusterMode=true \
+    -write.url=http://localhost:8480 \
+    -read.url=http://localhost:8481
+```
+
+With this configuration vmgateway will use the `vm_access` claim from the JWT token to restrict access to metrics.
+For example, if the JWT token contains the following `vm_access` claim:
+
+```json
+{
+  "vm_access": {
+    "tenant_id": {
+      "account_id": 0,
+      "project_id": 0
+    }
+  }
+}
+```
+
+Then vmgateway will proxy request to an endpoint with the following path:
+
+```console
+http://localhost:8480/select/0:0/
+```
+
+This allows to restrict access to specific tenants without having to create separate datasources in Grafana,
+or manually managing access at another proxy level. Moreover, using token claims such as `extra_labels`
+or `extra_filters` allows to further restrict access to specific metrics dynamically by using Identity Provider's user information.
```

Comment:
I think we should explicitly explain the thing with labels.
VM single node doesn't have muti-tenancy support, so that's the only way for it to get access management.
Hence, we should explicitly say how to get it for single or cluster versions

Reply:
Added separate sections for single-node and cluster explanation with more detailed explanation on how to use this for single-node installations.